arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,350 @@
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			relais_pins: relais_pins {
+				brcm,pins     = <28>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			button_pins: button_pins {
+				brcm,pins     = <23>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			tpm_pins: tpm_pins {
+				/* irq */
+				brcm,pins     = <22>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			rs485_0_pins: rs485_0_pins {
+				/* tx rx rts */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+			rs485_1_pins: rs485_1_pins {
+				/* tx rx rts */
+				brcm,pins     = <8 9 11>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&i2c6>;
+		__overlay__ {
+			// TODO: ANYTHING ELSE REQUIRED HERE?
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c6_gpio0>;
+			status = "okay";
+		};
+	};
+
+	fragment@3 {
+		target = <&i2c1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c1_gpio2>;
+			status = "okay";
+
+			expander: gpio@20 {
+				compatible = "ti,tca6416";
+				// TODO: which reg to use here?
+				reg = <0x20>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			rtc@51 {
+				compatible = "nxp,pcf2129";
+				// TODO: which reg to use here?
+				reg = <0x51>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@4 {
+		target = <&spi5>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 17 GPIO_ACTIVE_LOW>,
+				   <&gpio 18 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi5_pins &spi5_cs_pins>;
+			status = "okay";
+
+			tpm: tpm@0 {
+				compatible = "infineon,slb9670";
+				pinctrl-names = "default";
+				pinctrl-0 = <&tpm_pins>;
+				reg = <0>;
+				spi-max-frequency = <1000000>;	/* <= 43 MHz */
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <22 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+			ksz@1 {
+				compatible = "microchip,ksz8795";
+				reg = <1>;
+				// TODO: interrupt not to be needed? (not used in driver)
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <44 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+
+		};
+	};
+
+	fragment@5 {
+		target = <&spi6>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 6 GPIO_ACTIVE_LOW>,
+				   <&gpio 10 GPIO_ACTIVE_LOW>,
+				   <&gpio 27 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi6_pins &spi6_cs_pins>;
+			status = "okay";
+
+			aout: dac@0 {
+				compatible = "ti,dac7512";
+				reg = <0>;
+				spi-max-frequency = <5000000>;  /* <= 30 MHz */
```

Comment:
Well, I think the maximum speed *is* useful for documentation purposes because it avoids a lookup in the datasheet. (You might not have raised this point without the information.)